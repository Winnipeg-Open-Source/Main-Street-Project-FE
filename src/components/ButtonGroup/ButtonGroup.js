import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex } from 'pcln-design-system';

const Wrapper = styled(Flex)`
    & > :not(:last-child) {
        margin-right: 8px;
    }
`;

function ButtonGroup ({ className, children, ...props }) {
    return (
        <Wrapper className={className} {...props}>
            {children}
        </Wrapper>
    );
}

ButtonGroup.displayName = 'ButtonGroup';

ButtonGroup.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
};

ButtonGroup.defaultProps = {
    className: '',
};

export default memo(ButtonGroup);
