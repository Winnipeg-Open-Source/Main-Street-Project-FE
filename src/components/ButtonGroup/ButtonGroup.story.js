import React from 'react';
import { Button } from 'pcln-design-system';
import ButtonGroup from './ButtonGroup';

export const TwoButtons = () => (
    <ButtonGroup>
        <Button>One</Button>
        <Button>Two</Button>
    </ButtonGroup>
);

export const ThreeButtons = () => (
    <ButtonGroup>
        <Button>One</Button>
        <Button>Two</Button>
        <Button>Three</Button>
    </ButtonGroup>
);

export default {
    title: 'ButtonGroup',
    component: ButtonGroup,
};
