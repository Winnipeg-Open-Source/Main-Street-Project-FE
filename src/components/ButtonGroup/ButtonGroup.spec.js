import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import ButtonGroup from './ButtonGroup';
import { Button } from 'pcln-design-system';

describe('ButtonGroup', () => {
    it('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<ButtonGroup />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('has single button', () => {
        const { queryByText } = renderWithTheme(
            <ButtonGroup>
                <Button>First</Button>
            </ButtonGroup>
        );

        expect(queryByText(/First/i)).toBeInTheDocument();
        expect(queryByText(/Second/i)).not.toBeInTheDocument();
    });

    it('has multiple buttons', () => {
        const { getByText } = renderWithTheme(
            <ButtonGroup>
                <Button>First</Button>
                <Button>Second</Button>
            </ButtonGroup>
        );

        expect(getByText(/First/i)).toBeInTheDocument();
        expect(getByText(/Second/i)).toBeInTheDocument();
    });
});
