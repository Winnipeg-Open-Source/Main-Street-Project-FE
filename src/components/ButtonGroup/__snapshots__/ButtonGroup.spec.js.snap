// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC fCxHK"
  >
    <div
      class="sc-AxjAm sc-AxiKw sc-psDhf dCBSxT"
    />
  </div>
</DocumentFragment>
`;
